content/smyja/how-to-use-graphql-with-django.md

```diff
@@ -0,0 +1,436 @@
+
+```bash
+django-admin startproject restaurant_graphql_api
+cd restaurant_graphql_api
+```
+
+Next, we're going to create a new application within our project called `my_app` by running the following:
+
+```bash
+python manage.py startapp my_app
+```
+
+Then, we'll add `my_app` to our list of `INSTALLED_APPS` in our `settings.py` file under the `restaurant-graphql_api/` directory:
+
+```py
+INSTALLED_APPS = [
+  'my_app',
+  'django.contrib.admin',
+  'django.contrib.auth',
+  # ...
+]
+```
+
+## Step 3: using GraphQL with `graphene-django`
+
+To use GraphQL with Django, we will need to install the [`graphene-django`](https://docs.graphene-python.org/projects/django/en/latest/) package.
+
+```bash
+pip install graphene-django
+```
+
+This will help add GraphQL functionality to our restaurant Django app such as resolvers and mutations. Next, let's add `'graphene_django'` to the list of `INSTALLED_APPS` in our `settings.py` file:
+
+```py
+INSTALLED_APPS = [
+  'graphene_django',
+  'my_app',
+  'django.contrib.admin',
+  # ...
+]
+```
+
+Let's go to the `models.py` file in our project and then define a new `Restaurant` class:
+
+```py
+from django.db import models
+
+class Restaurant(models.Model):
+  name = models.CharField(max_length=100)
+  address = models.CharField(max_length=200)
+
+  def __str__(self):
+    return self.name
+```
+
+Inside the `Restaurant` class model above, we defined a few fields, `name` and `address`, along with a `__str__()` [dunder method](https://www.codecademy.com/resources/docs/python/dunder-methods) that returns the `name` of the restaurant.
+
+Next, let's register our new `Restaurant` model in the `admin.py` file of our application:
+
+```py
+from django.contrib import admin
+from . import models
+
+admin.site.register(models.Restaurant)
+```
+
+It is now time to create and perform a migration for this new data. This will allow our `Restaurant` model to be referenced in a GraphQL schema (which we will define later on). To make the migration, we can run `python manage.py makemigrations`; to apply the migrations, let's run `python manage.py migrate`.
+
+By now, we may encounter the following error:
+
+```shell
+ImportError: cannot import name 'force_text' from 'django.utils.encoding
+```
+
+The `ImportError` is due to Django 4.0 not supporting the `force_text` variable (which the `graphene` package uses with earlier versions of Django). To resolve this, we can add the following to our `settings.py` file:
+
+```py
+import django
+from django.utils.encoding import force_str
+
+django.utils.encoding.force_text = force_str
+```
+
+Alternatively, we can downgrade our Django version to 3.2.x. At this point, it would be good to run `python manage.py runserver` and check `http://127.0.0.1:8000` on a browser to ensure our application starts properly.
```

Comment:
```suggestion
Alternatively, we can downgrade our Django version to 3.2.x. 

After this, it would be good to run `python manage.py runserver` and check `http://127.0.0.1:8000` on a browser to ensure our application starts properly.
```